### PR TITLE
example server: support serving over stdio

### PR DESCRIPTION
**Pull request by gerow**
This chnage adds an "stdio" flag to the example server in order to
support `remote` p11-kit module entries, which dynamically launch a
managed server for clients that support it. This includes clients that
use gnutls.

See pkcs11.conf(5) for details on what the remote entry is meant to do.

This is still a WIP as I think gnutls and p11-kit in managed mode still have trouble accessing go-p11-kit remote modules when configured in this manner.

### CODE REVIEW

**Comment by gerow** [comment]
Using my example config it seems that `p11tool` can only see one of the objects:

```
$ cat /usr/share/p11-kit/modules/gop11kit.module 
remote: |/home/gerow/repos/go-p11-kit/bin/example-p11-kit-server --priv /home/gerow/repos/go-p11-kit/example/priv.pem --pub /home/gerow/repos/go-p11-kit/example/pub.pem --cert /home/gerow/repos/go-p11-kit/example/cert.pem --stdio
```

```
$ p11tool --list-all
warning: no token URL was provided for this operation; the available tokens are:

pkcs11:model=p11-kit-trust;manufacturer=PKCS%2311%20Kit;serial=1;token=System%20Trust
pkcs11:model=example-server;manufacturer=go-p11-kit;serial=12345678;token=example
2022/04/27 16:41:37 Handling over stdio: read request: reading request header: EOF
```

And it seems to complain even then. In addition, trying to use the remote module using `gnutls` complains about missing functions it probably uses to determine if the token needs a pin.

```
$ gnutls-cli google.com --app-proto=https --x509keyfile='pkcs11:model=example-server;manufacturer=go-p11-kit;serial=12345678;token=example' --x509certfile='pkcs11:model=example-server;manufacturer=go-p11-kit;serial=12345678;token=example'
Processed 141 CA certificate(s).
2022/04/27 16:42:49 Error with C_GetSessionInfo: CKR_FUNCTION_NOT_SUPPORTED
Token 'example' with URL 'pkcs11:model=example-server;manufacturer=go-p11-kit;serial=12345678;token=example' requires user PIN
Enter PIN: 
2022/04/27 16:42:52 Error with C_Login: CKR_FUNCTION_NOT_SUPPORTED
*** Error loading cert file.
2022/04/27 16:42:52 Handling over stdio: read request: reading request header: EOF
```

I also tried to add `pin-source=file:/dev/null` to the pkcs11 URIs, but that didn't seem to make gnutls any happier.

File: example/example-p11-kit-server/main.go [hunk 1]
```diff
@@ -35,9 +36,10 @@ An example p11-kit server that can serve on disk files as a PKCS #11 module.
 
 Flags:
 
-    --cert Path to a file containing a PEM encoded certificate.
-    --priv Path to a file containing a PEM encoded private key.
-    --pub  Path to a file containing a PEM encoded public key.
+    --cert  Path to a file containing a PEM encoded certificate.
+    --priv  Path to a file containing a PEM encoded private key.
+    --pub   Path to a file containing a PEM encoded public key.
+	--stdio Serve a single connection over stdio, intended for p11-kit's 'remote' module entries.
```
Comment (ericchiang) [inline example/example-p11-kit-server/main.go:42]: nit: the indentation uses spaces not tags

**Review by ericchiang — APPROVED** [review]

**Comment by gerow** [comment]
OK cool, I guess this doesn't need to be a WIP after all. The gnutls things are probably their own problem, opened #21 to try and address that, but this is probably fine as-is.

**Comment by ericchiang** [comment]
lgtm, @gerow I've invited you to be a maintainer so feel free to merge :)

**Pull request merged**